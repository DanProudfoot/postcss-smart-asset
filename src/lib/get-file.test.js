const fs = require('fs');
const path = require('path');
const getFile = require('./get-file');

describe('get-file', () => {
    const fileContent = fs.readFileSync('test/fixtures/pixel.gif');
    const dir = { from: 'test/fixtures' };
    const warn = (message) => assert.fail(null, null, message);

    test('should read file without basePath option', () => {
        const asset = {
            pathname: '../pixel.gif',
            absolutePath: 'test/fixtures/pixel.gif'
        };
        const file = getFile(asset, {}, dir, warn);

        assert.deepEqual(file, {
            path: 'test/fixtures/pixel.gif',
            contents: fileContent,
            mimeType: 'image/gif'
        });
    });

    test('should show warn message when can\'t read file', () => {
        const asset = {
            pathname: '../pixel.gif',
            absolutePath: 'test/fixtures/pixel-no-exists.gif'
        };
        let warnMessage = false;

        getFile(asset, {}, dir, (message) => warnMessage = message);

        assert.equal(
            warnMessage,
            'Can\'t read file \'test/fixtures/pixel-no-exists.gif\', ignoring'
        );
    });

    test('should read file with basePath option', () => {
        const options = { basePath: path.resolve('test/fixtures/imported') };
        const asset = {
            pathname: '../pixel.gif',
            absolutePath: 'test/fixtures/pixel-not-exists.gif'
        };
        const file = getFile(asset, options, dir, warn);

        assert.deepEqual(file, {
            path: path.resolve('test/fixtures/pixel.gif'),
            contents: fileContent,
            mimeType: 'image/gif'
        });
    });

    test('should read file with multiple basePath option', () => {
        const options = { basePath: [
            'not-exists-folder',
            path.resolve('test/fixtures/imported')
        ] };
        const asset = {
            pathname: '../pixel.gif',
            absolutePath: 'test/fixtures/pixel-not-exists.gif'
        };
        const file = getFile(asset, options, dir, warn);

        assert.deepEqual(file, {
            path: path.resolve('test/fixtures/pixel.gif'),
            contents: fileContent,
            mimeType: 'image/gif'
        });
    });
});
