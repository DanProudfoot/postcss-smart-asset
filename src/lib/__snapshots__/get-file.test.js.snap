// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-file should read file with basePath option 1`] = `
Object {
  "mimeType": "image/gif",
  "path": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/pixel.gif",
}
`;

exports[`get-file should read file with multiple basePath option 1`] = `
Object {
  "mimeType": "image/gif",
  "path": "/Users/sebastian/Workspace/sebastian-software/postcss-smart-asset/test/fixtures/pixel.gif",
}
`;

exports[`get-file should read file without basePath option 1`] = `
Object {
  "mimeType": "image/gif",
  "path": "test/fixtures/pixel.gif",
}
`;
